gpuallocator/device.go

```diff
@@ -7,61 +7,99 @@ import (
 	"sort"
 	"strings"
 
-	"github.com/NVIDIA/gpu-monitoring-tools/bindings/go/nvml"
+	"github.com/NVIDIA/go-gpuallocator/internal/links"
+	"gitlab.com/nvidia/cloud-native/go-nvlib/pkg/nvlib/device"
+	"gitlab.com/nvidia/cloud-native/go-nvlib/pkg/nvml"
 )
 
 // Device represents a GPU device as reported by NVML, including all of its
 // Point-to-Point link information.
 type Device struct {
-	*nvml.Device
+	device.Device
 	Index int
 	Links map[int][]P2PLink
+	// We cache certain values for the device.
+	// Deprecated: Use the device.Device methods instead.
+	UUID string
+	// Deprecated: Use the device.Device methods instead.
+	PCI struct {
+		BusID string
+	}
+	CPUAffinity *uint
```

Comment:
Should `CPUAffinity` or `GetCPUAffinity` be a method on the `device.Device` interface instead?